legend-sdlc-generation-file/src/test/java/org/finos/legend/sdlc/generation/artifact/TestArtifactGenerationFactory.java

```diff
@@ -0,0 +1,84 @@
+        {
+            throw new RuntimeException("Unable to build PureModelContextData with path '" + path + "'", e);
+        }
+    }
+
+    @Test
+    public void testArtifactGenerationFactory()
+    {
+        PureModelContextData data = getPureModelContextDataFromPath("ArtifactGenerationFactoryTestData.json");
+        PureModel pureModel = new PureModel(data, null, DeploymentMode.TEST);
+        ArtifactGenerationFactory factory = new ArtifactGenerationFactory(pureModel, data, data.getElements());
+        Assert.assertEquals(2, factory.getExtensions().size());
+        TestSimpleFunctionArtifactGenerationExtension functionArtifactGenerationExtension = LazyIterate.selectInstancesOf(factory.getExtensions(), TestSimpleFunctionArtifactGenerationExtension.class).getFirst();
+        TestSimpleEnumArtifactGenerationExtension enumArtifactGenerationExtension =  LazyIterate.selectInstancesOf(factory.getExtensions(), TestSimpleEnumArtifactGenerationExtension.class).getFirst();
+        MutableMap<ArtifactGenerationExtension, List<ArtifactGenerationResult>> results = factory.generate();
+        Assert.assertEquals(2, results.size());
+        // check enum generation extension
+        List<ArtifactGenerationResult> enumFullResults =  results.get(enumArtifactGenerationExtension);
+        Assert.assertEquals(1, enumFullResults.size());
+        ArtifactGenerationResult enumResult = enumFullResults.get(0);
+        Assert.assertEquals(TestSimpleEnumArtifactGenerationExtension.class, enumResult.getGenerator().getClass());
+        Assert.assertEquals(enumResult.getElement(), pureModel.getPackageableElement("model::MyEnum"));
+        Assert.assertEquals(1, enumResult.getResults().size());
+        GenerationOutput enumOutput = enumResult.getResults().get(0);
+        Assert.assertEquals("Some output for enumeration 'MyEnum'", enumOutput.getContent());
+        Assert.assertEquals("txt", enumOutput.getFormat());
+        Assert.assertEquals("SomeTestOutput.txt", enumOutput.getFileName());
+        // check function generation results
+        List<ArtifactGenerationResult> functionResults =  results.get(functionArtifactGenerationExtension);
+        Assert.assertEquals(1, functionResults.size());
+        ArtifactGenerationResult functionResult = functionResults.get(0);
+        Assert.assertEquals(TestSimpleFunctionArtifactGenerationExtension.class, functionResult.getGenerator().getClass());
+        Assert.assertEquals(functionResult.getElement(), pureModel.getPackageableElement("model::myFunction_String_1__String_1_"));
+        Assert.assertEquals(1, functionResult.getResults().size());
+        GenerationOutput functionOutput = functionResult.getResults().get(0);
+        Assert.assertEquals("Some output for function 'myFunction_String_1__String_1_'", functionOutput.getContent());
+        Assert.assertEquals("txt", functionOutput.getFormat());
+        Assert.assertEquals("MyFunction.txt", functionOutput.getFileName());
+    }
+
+}
```

Comment:
```suggestion
    }
}
```

Reply:
apologizes wanted them all to be sync but put them all in sync in the wrong format :)